ibmcloud/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,212 @@
+╷
+│ Error: Iteration over null value
+│
+│   on main.tf line 16, in locals:
+│   15:   is_policies_and_roles = flatten([
+│   16:     for policy in data.ibm_iam_user_policy.user_policies.policies: [
+│   17:       for resource in policy.resources: policy.roles
+│   18:       if resource.service == "is" && resource.resource_group_id == "" && resource.resource_instance_id == ""
+│   19:     ]
+│   20:   ])
+│     ├────────────────
+│     │ data.ibm_iam_user_policy.user_policies.policies is null
+│
+│ A null value cannot be used as the collection in a 'for' expression.
+
+For more information, please see the following Slack threads: -
+```
+
+This issue was observed with various releases of the [terraform-provider-ibm](https://github.com/IBM-Cloud/terraform-provider-ibm), including `1.34.0`, `1.41.1` and `1.42.0-beta0`.
+
+The issue was also observed with a very simple Terraform playbook: -
+
+```code
+locals {
+  is_policies_and_roles = flatten([
+    for policy in data.ibm_iam_user_policy.user_policies.policies: [
+    ]
+  ])
+}
+
+data "ibm_iam_user_policy" "user_policies" {
+  ibm_id = var.ibmcloud_user_id
+}
+```
+
+and highlighted the fact that Terraform was unable to retrieve the required User Policies via `data.ibm_iam_user_policy.user_policies.policies` in order to be able to loop across the returned list.
+
+Further investigation suggested that the issue only occurred when the IBM Cloud account had a specific User Policy, as per the following example: -
+
+```text
+Policy ID:   7152fe85-36da-46b5-ab9b-1eac8d212cd5
+Roles:       Reader
+Resources:
+             Service Name   containers-kubernetes
+             Region         us-south
+             namespace      default
+```
+
+If this policy was removed from the account, the plan ran to completion, and the engineer was able to then run `terraform apply`.
+
+If the policy was then added back into the account, the same `A null value cannot be used as the collection in a 'for' expression` exception occurred.
+
+After discussions with the community who maintain the [terraform-provider-ibm](https://github.com/IBM-Cloud/terraform-provider-ibm) project, an issue - [Data source of IAM policies failing to list the policies if the policy has service specific attributes #3801](https://github.com/IBM-Cloud/terraform-provider-ibm/issues/3801) - was raised, and is being actively worked at time of writing ( May 2022 ).
+
+In the meantime, if this problem occurs for other users, a check for policies containing service-specific attributes should be undertaken, as per the above example, where the User Policy is tagged with a Service Name of `container-kubernetes`.
+
+Whilst it's not 100% clear which service-specific attributes might cause this issue, one can check using the [IBM Cloud CLI tool](https://cloud.ibm.com/docs/cli?topic=cli-getting-started) and the `vpc-infrastructure` plugin ( aka `is` ), using a command similar to the following example: -
```

Comment:
I don't think the `ibmcloud iam` check is using the `vpc-infrastructure` plugin, because it's using an `iam` command rather than `is`. E.g., this worked for me:

```
$ ibmcloud plugin uninstall vpc-infrastructure
Uninstalling plug-in 'vpc-infrastructure'...
OK
Plug-in 'vpc-infrastructure' was successfully uninstalled.
$ ibmcloud iam user-policies MATTARNO@uk.ibm.com
Getting policies of user MATTARNO@uk.ibm.com under current account as MATTARNO@uk.ibm.com...
OK
...
```

Reply:
Good spot, and I confirmed this on a new Fyre box, with no plugins installed - `ibmcloud iam` commands are built into the CLI directly. Have removed it from the doc, with the most recent commit